tests/test_url.py

```diff
@@ -553,6 +553,69 @@ def test_canonicalize_url_idna_exceptions(self):
                     label=u"example"*11))
 
 
+class DataURITests(unittest.TestCase):
+
+    def test_default_mediatype_charset(self):
+        result = parse_data_uri("data:,A%20brief%20note")
+        self.assertEqual(result.media_type, "text/plain")
+        self.assertEqual(result.media_type_parameters, {"charset": "US-ASCII"})
+        self.assertEqual(result.data, b"A brief note")
+
+    def test_text_uri(self):
+        result = parse_data_uri(u"data:,A%20brief%20note")
+        self.assertEqual(result.data, b"A brief note")
+
+    def test_bytes_uri(self):
+        result = parse_data_uri(b"data:,A%20brief%20note")
+        self.assertEqual(result.data, b"A brief note")
+
+    def test_non_ascii_uri(self):
+        with self.assertRaises(UnicodeEncodeError):
+            parse_data_uri(u"data:,é")
```

Comment:
I think raising UnicodeEncodeError is not a good way to handle it in Python 3. We settled on unicode URLs by default for Python 3, they shouldn't raise errors. I tried it in Firefox - it escaped this URL to `data:,%C3%A9` and then processed it. `w3lib.url.safe_url_string` function escaped URL the same way. I'm not 100% sure about that, but it looks like we should use safe_url_string here instead of encoding to ascii. This way data: url will be handled the same way as a browser handles them. What do you think?